api/src/controllers/validation.ts

```diff
@@ -1,51 +1,92 @@
 import { HttpError } from "./error";
-import { isInArray } from "../helpers";
+import { isInSet, not } from "../helpers";
+import { StringLiteral } from "../types";
 
-type StringLiteral<T> = T extends string
-  ? string extends T
-    ? never
-    : T
-  : never;
+type NonEmptyArray<T> = [T, ...T[]];
```

Comment:
TIL!

(I've wondered for a while why Prismic's `RichTextField` type is defined as `[] | [RTNode, ...RTNode[]]`, and now it makes sense.)

Reply:
Yeah once you see it it's quite an elegant way of expressing this imo!